extensions/wikia/Recirculation/js/experiments/mix.js

```diff
@@ -0,0 +1,135 @@
+/*global require*/
+require([
+	'jquery',
+	'wikia.window',
+	'wikia.abTest',
+	'wikia.log',
+	'ext.wikia.recirculation.tracker',
+	'ext.wikia.recirculation.utils',
+	'ext.wikia.recirculation.discussions',
+	'ext.wikia.recirculation.tracker',
+	require.optional('videosmodule.controllers.rail')
+], function(
+	$,
+	w,
+	abTest,
+	log,
+	tracker,
+	utils,
+	discussions,
+	tracker,
+	videosModule
+) {
+	/**
+	 *
+	 * This test is setup using the AB testing tool.
+	 * We inject a variable called recircExperiment that dictates which content
+	 * gets rendered in which placement. The structure of the variable is:
+	 *
+	 *  var recircExperiment = [
+	 *  	{
+	 *  		id: 'unique identifier - only used if you want to use data across multiple views',
+	 *  		sorce: 'the id you want to use for data - only used if you want to use data across multiple views',
+	 *  		placement: 'name of the view',
+	 *  		helper: 'name of the helper',
+	 *  		options: {
+	 *  			any options to pass to the helper
+	 *  		}
+	 *  	},
+	 *  	{
+	 *  		placement: 'impactFooter',
+	 *  		helper: 'cakeRelatedContent',
+	 *  		options: {
+	 *  			limit: 21
+	 *  		}
+	 *  	}
+	 *  ];
+	 *
+	 */
+
+	var recircExperiment = w.recircExperiment || false,
+		experimentName = 'RECIRCULATION_MIX',
+		logGroup = 'ext.wikia.recirculation.experiments.mix',
+		group = abTest.getGroup(experimentName),
+		views = {},
+		saved = {};
+
+	if (!recircExperiment || w.wgContentLanguage !== 'en') {
+		if (videosModule) {
+			videosModule('#RECIRCULATION_RAIL');
+		}
+		return;
+	}
+
+	recircExperiment.forEach(function(experiment, index) {
+		var deferred = $.Deferred();
+
+		views[experiment.placement] = views[experiment.placement] || [];
+
+		if (experiment.id) {
+			saved[experiment.id] = deferred;
+		}
+
+		if (experiment.helper) {
+			var helperString = 'ext.wikia.recirculation.helpers.' + experiment.helper;
+
+			require([helperString], function (helper) {
+				helper(experiment.options).loadData()
+					.then(function(data) {
+						deferred.resolve(data);
+					});
+			});
+		}
+
+		if (experiment.source && saved[experiment.source]) {
+			saved[experiment.source].then(function(data) {
+				deferred.resolve({
+					title: data.title,
+					items: data.items.slice(experiment.options.offset)
+				});
+			});
+		}
+
+		views[experiment.placement].push(deferred);
+	});
+
+	$.each(views, function (key, value) {
+		var viewString = 'ext.wikia.recirculation.views.' + key;
+
+		log('Initializing View: ' + key, 'info', logGroup);
+		require([viewString], function (viewFactory) {
+			$.when.apply($, value)
+				.then(function () {
+					var view = viewFactory(),
+						args = Array.prototype.slice.call(arguments),
+						data = {
+							title: '',
+							items: []
+						};
+
+					log(args, 'info', logGroup);
+
+					args.forEach(function (result, index) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the unused function parameter "index". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)